pkg/cmd/miscellaneous.go

```diff
@@ -350,17 +367,72 @@ func getRole() string {
 }
 
 /*
-getTechnicalID returns the Technical Id, which used as clustername of the shoot cluster
+getTargetMapInfo retun garden,project,seed,shoot,shootTechnicalID to global getTargetInfo
+*/
+func getTargetMapInfo() {
+	if len(getTargetInfo) == 0 {
+		var target Target
+		ReadTarget(pathTarget, &target)
+		for _, t := range target.Stack() {
+			getTargetInfo[string(t.Kind)] = string(t.Name)
+		}
+
+		shootName := getTargetInfo["shoot"]
+		project, err := getProjectObject()
+		checkError(err)
+		shoot, err := getShootObject(*project.Spec.Namespace, shootName)
+		checkError(err)
+		getTargetInfo["shootTechnicalID"] = shoot.Status.TechnicalID
+
+		// user for case garden-project-shoot add [seed] = seedName
+		if isTargeted("project", "shoot") {
+			getTargetInfo["seed"] = *shoot.Spec.SeedName
+		}
+	}
+}
+
+/*
+lookup Target Kind ("kind value") return Target Name "name value" from target file ~/.garden/sessions/plantingSession/target
+*/
+func getTargetName(Kind string) (string, error) {
+	var target Target
+	ReadTarget(pathTarget, &target)
+	for _, t := range target.Stack() {
+		if string(t.Kind) == Kind {
+			return string(t.Name), nil
+		}
+	}
+	return "", errors.New("Kind:" + Kind + "no found")
+}
+
+/*
+check if target Kind is exist in target file ~/.garden/sessions/plantingSession/target
 */
-func getTechnicalID() string {
+func isTargeted(args ...string) bool {
 	var target Target
 	ReadTarget(pathTarget, &target)
 
-	gardenClientset, err := target.GardenerClient()
-	checkError(err)
-	project, err := gardenClientset.CoreV1beta1().Projects().Get(target.Stack()[1].Name, metav1.GetOptions{})
-	checkError(err)
-	shoot, err := gardenClientset.CoreV1beta1().Shoots(*project.Spec.Namespace).Get(target.Stack()[2].Name, metav1.GetOptions{})
-	checkError(err)
-	return shoot.Status.TechnicalID
+	targetMap := make(map[string]interface{})
+	for _, t := range target.Stack() {
+		targetMap[string(t.Kind)] = string(t.Name)
+	}
+
+	for _, t := range args {
+		if _, ok := targetMap[t]; ok {
+		} else {
+			return false
+		}
+	}
+
+	return true
+}
+
+func isValidTargetInfo(key ...string) error {
+	for _, t := range key {
+		if _, ok := getTargetInfo[t]; ok {
+			return nil
+		}
+	}
+
+	return errors.New("key no found")
```

Comment:
not needed `isValidTargetInfo`, can be removed when implementing `getFromTargetInfo`

Reply:
fixed